### ISSUE TITLE
Forbid old-styled classes

### ISSUE THREAD

**Issue by sobolevn** [wemake-services/wemake-python-styleguide#35]
In `python2` we had different classes: `class Some:` and `class Some(object):` were totally different things.

Currently, we don't have this thing in `python3`, but since these times I believe that  new-styled classes are still better.

1. It has more semantics: this class is related to `object`
2. It has unified interface: `class Base(object)` and `class Child(Base)` vs `class Base` and `class Child(Base)`
3. Explicit is better than implicit 

So, I believe we should ban classes with old-styled base class.
General rule is: **always write super-classes' names**

## Correct

```python
class Philosopher(object):
    def __init_subclass__(cls, default_name, **kwargs):
        ...

class AustralianPhilosopher(Philosopher, default_name="Bruce"):
    ...

class WithMetaClass(object, metaclass=MyMeta):
   ...
```

## Wrong

```python
class Philosopher: ...
class AustralianPhilosopher(default_name="Bruce"): ...
class WithMetaClass(metaclass=MyMeta): ...
```

**Comment by AlwxSin** [comment]
![ezgif-2-dc2f4987da](https://user-images.githubusercontent.com/3910216/42565639-ae4d2e1e-850c-11e8-9ed1-16f92611e25f.gif)

**Comment by AlwxSin** [comment]
My point is that `Class(object)` style is legacy.
Every good python developer should know that in python everything comes from object. Otherwise he is not that good.
Why should you write about that explicitly?

**Comment by sobolevn** [comment]
That's not about the technical aspect. It is about consistency.

We have projects that share both `class Some:` and `class Some(object):` in the code base.
And that's inconsistent. 

To fix that we need to stick to a single writing style. We have a choice here: new style vs old style.
And that's where I prefer old style better. By the reasons I have provided in the description.

So, now we will have consistent styles across the class definitions. Which is a good thing for us.

**Comment by sobolevn** [comment]
> Both are subclasses

**Comment by AlwxSin** [comment]
> Yes, both classes are subclasses

And good developer knows that.
But you cannot clearly see the difference between these classes though.
